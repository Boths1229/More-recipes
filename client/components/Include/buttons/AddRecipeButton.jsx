import React from 'react';
import PropTypes, { any } from 'prop-types';
import AddModal from '../../Include/modals/AddModal';

/**
 * @description functional component to render the add recipe button
 *
 * @param {props} props
 *
 * @returns {JSX} return JSX
 */
const AddRecipeButton = (props) => {
  const {
    onChange, addRecipe, value, saveImageToCloud, recipeImage
  } = props;
  return (
    <div className="row header-nav ">
      <div className="col-sm-6 col-6">
        <p className="pt-3 font">Add New recipe</p>
      </div>
      <div className="col-sm-6 col-6 pt-2 pb-2">
        <button
          type="button"
          className="btn rfloat mt-2"
          data-toggle="modal"
          data-target="#exampleModal"
        >
          <i className="fa fa-plus-square-o" aria-hidden="true" />
        </button>
        <AddModal
          onChange={onChange}
          addRecipe={addRecipe}
          value={value}
          saveImageToCloud={saveImageToCloud}
          recipeImage={recipeImage}
        />
      </div>
    </div>
  );
};

AddRecipeButton.propTypes = {
  onChange: PropTypes.func.isRequired,
  addRecipe: PropTypes.func.isRequired,
  value: PropTypes.objectOf(any).isRequired,
};

export default AddRecipeButton;
