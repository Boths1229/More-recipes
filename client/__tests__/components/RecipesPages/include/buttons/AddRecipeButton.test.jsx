import React from 'react';
import { shallow } from 'enzyme';
import AddRecipeButton from '../../../../../components/Include/buttons/AddRecipeButton';

describe('Add recipe button component', () => {
  it('matches snapshot', () => {
    const output = shallow(<AddRecipeButton />);
    expect(output).toMatchSnapshot();
  });
});
