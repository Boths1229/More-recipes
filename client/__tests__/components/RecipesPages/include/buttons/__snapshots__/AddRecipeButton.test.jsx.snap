// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add recipe button component matches snapshot 1`] = `
<div
  className="row header-nav "
>
  <div
    className="col-sm-6 col-6"
  >
    <p
      className="pt-3 font"
    >
      Add New recipe
    </p>
  </div>
  <div
    className="col-sm-6 col-6 pt-2 pb-2"
  >
    <button
      className="btn rfloat mt-2"
      data-target="#exampleModal"
      data-toggle="modal"
      type="button"
    >
      <i
        aria-hidden="true"
        className="fa fa-plus-square-o"
      />
    </button>
    <AddModal />
  </div>
</div>
`;
